### ISSUE TITLE
"yarn start" fails on Node 7

### ISSUE THREAD

**Issue by ZenSoftware** [graphql-dotnet/graphql-dotnet#215]
`yarn start` fails with an **Unhandled 'error' event**

**System configuration**
graphql-dotnet master branch v0.13.1.601
Windows 10 Home v10.0.14393 Build 14393
node v7.1.0
npm v3.10.9
yarn@0.17.2

**Sequence of commands executed:**
```
git clone --depth 1 https://github.com/graphql-dotnet/graphql-dotnet.git
cd graphql-dotnet
yarn install
npm run build
yarn start
```

All commands execute successfully, besides `yarn start`.  The full log of my command prompt of the above being executed can be viewed here: 
[command-log.txt](https://github.com/graphql-dotnet/graphql-dotnet/files/593789/command-log.txt)

If any further information is needed, please do let me know.

Specifically, here is the part of the log that describes the error:
```
$ yarn start
yarn start v0.17.2
$ webpack --progress                events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: write EINVAL
    at exports._errnoException (util.js:1022:11)
    at WriteStream.Socket._writeGeneric (net.js:715:26)
    at WriteStream.Socket._write (net.js:734:8)
    at doWrite (_stream_writable.js:334:12)
    at writeOrBuffer (_stream_writable.js:320:5)
    at WriteStream.Writable.write (_stream_writable.js:247:11)
    at WriteStream.Socket.write (net.js:661:40)
    at D:\graphql-dotnet\node_modules\webpack\bin\convert-argv.js:421:20
    at Compiler.<anonymous> (D:\graphql-dotnet\node_modules\webpack\lib\ProgressPlugin.js:77:4)
    at Compiler.applyPlugins (D:\graphql-dotnet\node_modules\tapable\lib\Tapable.js:26:37)
    at Compiler.<anonymous> (D:\graphql-dotnet\node_modules\webpack\lib\Compiler.js:193:12)
    at Compiler.emitRecords (D:\graphql-dotnet\node_modules\webpack\lib\Compiler.js:282:37)
    at Compiler.<anonymous> (D:\graphql-dotnet\node_modules\webpack\lib\Compiler.js:187:11)
    at D:\graphql-dotnet\node_modules\webpack\lib\Compiler.js:275:11
    at Compiler.applyPluginsAsync (D:\graphql-dotnet\node_modules\tapable\lib\Tapable.js:60:69)
    at Compiler.afterEmit (D:\graphql-dotnet\node_modules\webpack\lib\Compiler.js:272:8)
    at Compiler.<anonymous> (D:\graphql-dotnet\node_modules\webpack\lib\Compiler.js:267:14)
    at D:\graphql-dotnet\node_modules\async\lib\async.js:52:16
    at done (D:\graphql-dotnet\node_modules\async\lib\async.js:246:17)
    at D:\graphql-dotnet\node_modules\async\lib\async.js:44:16
    at D:\graphql-dotnet\node_modules\graceful-fs\graceful-fs.js:43:10
    at FSReqWrap.oncomplete (fs.js:111:15)
error Command failed with exit code 1.
```

**Comment by joemcbride** [comment]
This seems to be an issue with the new Yarn version, perhaps in combination with the latest node?  How did you install Yarn?  Was it via `npm install -g yarn`?

This setup currently works for me:

node v6.9.1
npm v3.10.6
yarn v0.17.2 (installed via msi)

**Comment by ZenSoftware** [comment]
Yes that is correct.  I installed yarn via `npm install -g yarn`

**Comment by ZenSoftware** [comment]
I uninstalled yarn via `npm uninstall -g yarn` and then installed yarn via the [Winows msi package](https://yarnpkg.com/en/docs/install).  I went ahead and `git clone` the repo again and went through the sequence of commands described in the original post. yarn is now **v0.16.1** after the msi install.

After `yarn install` there seems to be a bit more of a messy error in regards to optional dependencies, I am presuming it is irrelevant.  `yarn start` will go ahead and start the webpack processing, but fails again in the same manner.  If I may speculate, maybe it is something failing in the webpack bundling, rather than yarn.  Although I guess webpack would fail due to something funky going on in how yarn brings down node_modules.  Sorry I am not sure how to investigate further.  I have only stumbled across this repo today, but would love to support it.

**System configuration this time**
Windows 10 Home v10.0.14393 Build 14393
node v7.1.0
npm v3.10.9
**yarn v0.16.1**

If you are interested, here is the log for the command prompt for this particular configuration.  Everything is the same besides the optional dependency error, which I am presuming is irrelevant (which did not happen with my original version of yarn).
[command-log-yarn0-16-1.txt](https://github.com/graphql-dotnet/graphql-dotnet/files/593970/command-log-yarn0-16-1.txt)

**Comment by joemcbride** [comment]
Thanks for doing some investigation!  Yes, there were some optional dependency issues with Yarn 0.16.1 on windows.  My next guess would be the node version?  I'm honestly not sure.  Our CI build started to fail with the latest version of Yarn as well, so there is some combination of versions that node + yarn + windows aren't playing nice together.

You can also download the latest 0.17.2 msi from from the project directly - https://github.com/yarnpkg/yarn/releases

Either way, Yarn is not required to be used.  You should be able to use `npm install` and `npm start` instead of Yarn and be okay.

**Comment by ZenSoftware** [comment]
Alright! I managed to get `yarn start` to works on node v6.9.1!  

**Final successful combination for my system:**
graphql-dotnet master branch v0.13.1.601
Windows 10 Home v10.0.14393 Build 14393
node v6.9.1
npm v3.10.9
yarn v0.17.2 

I tested that both the MSI package for yarn, and yarn installed via npm, both fix the issue if using node v6.9.1.  Neither yarn package works using node v7.1.0.  Hopefully it will help you narrow down what is going wrong.  Cheers my friend @joemcbride
